docs/src/embeddings/custom_embedding_function.md

```diff
@@ -15,198 +15,225 @@ There is another optional layer of abstraction available: `TextEmbeddingFunction
 
 Let's implement `SentenceTransformerEmbeddings` class. All you need to do is implement the `generate_embeddings()` and `ndims` function to handle the input types you expect and register the class in the global `EmbeddingFunctionRegistry`
 
-```python
-from lancedb.embeddings import register
-from lancedb.util import attempt_import_or_raise
-
-@register("sentence-transformers")
-class SentenceTransformerEmbeddings(TextEmbeddingFunction):
-    name: str = "all-MiniLM-L6-v2"
-    # set more default instance vars like device, etc.
-
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self._ndims = None
-    
-    def generate_embeddings(self, texts):
-        return self._embedding_model().encode(list(texts), ...).tolist()
-
-    def ndims(self):
-        if self._ndims is None:
-            self._ndims = len(self.generate_embeddings("foo")[0])
-        return self._ndims
-
-    @cached(cache={}) 
-    def _embedding_model(self):
-        return sentence_transformers.SentenceTransformer(name)
-```
-
-This is a stripped down version of our implementation of `SentenceTransformerEmbeddings` that removes certain optimizations and defaul settings.
+
+=== "Python"
+
+    ```python
+    from lancedb.embeddings import register
+    from lancedb.util import attempt_import_or_raise
+
+    @register("sentence-transformers")
+    class SentenceTransformerEmbeddings(TextEmbeddingFunction):
+        name: str = "all-MiniLM-L6-v2"
+        # set more default instance vars like device, etc.
+
+        def __init__(self, **kwargs):
+            super().__init__(**kwargs)
+            self._ndims = None
+
+        def generate_embeddings(self, texts):
+            return self._embedding_model().encode(list(texts), ...).tolist()
+
+        def ndims(self):
+            if self._ndims is None:
+                self._ndims = len(self.generate_embeddings("foo")[0])
+            return self._ndims
+
+        @cached(cache={})
+        def _embedding_model(self):
+            return sentence_transformers.SentenceTransformer(name)
+    ```
+
+=== "TypeScript"
+
+    ```ts
+    --8<--- "nodejs/examples/custom_embedding_function.ts:imports"
+    --8<--- "nodejs/examples/custom_embedding_function.ts:embedding_impl"
```

Comment:
It might be nice to have some whitespace here:
```suggestion
    --8<--- "nodejs/examples/custom_embedding_function.ts:imports"
    
    --8<--- "nodejs/examples/custom_embedding_function.ts:embedding_impl"
```